peps/management/commands/generate_pep_pages.py

```diff
@@ -51,16 +51,19 @@ def verbose(msg):
         verbose("== Starting PEP page generation")
 
         with ExitStack() as stack:
-            verbose(f"== Fetching PEP artifact from {settings.PEP_ARTIFACT_URL}")
-            temp_file = self.get_artifact_tarball(stack)
-            if not temp_file:
-                verbose("== No update to artifacts, we're done here!")
-                return
-            temp_dir = stack.enter_context(TemporaryDirectory())
-            tar_ball = stack.enter_context(TarFile.open(fileobj=temp_file, mode='r:gz'))
-            tar_ball.extractall(path=temp_dir, numeric_owner=False)
-
-            artifacts_path = os.path.join(temp_dir, 'peps')
+            if settings.PEP_REPO_PATH is not None:
+                artifacts_path = settings.PEP_REPO_PATH
```

Comment:
This if block doesn’t need to be in the exit stack context.

Reply:
The else clause of the if statement uses `stack` so I can't just move the entire if-statement up and out of the stack.

What kind of change were you thinking of exactly?

Reply:
Have the with block inside the else block

Reply:
Unfortunately that doesn't work. You'll notice that the exit stack is used to clean up files decompressed to `artifacts_path` which is used by the paragraphs following the if-statement. If I limit the scope of the exit stack to the else-clause it deletes the decompressed files before the rest of the command can read them.

And indeed, making that change anyway caused `python3 manage.py test peps` to start failing.

So I don't recommend moving the ExitStack around. Is there a particular concern of yours for having a single extra variable assignment inside the ExitStack that was there before?

Reply:
Ah I hadn’t noticed the uses of `stack` so far down!